src/circular_buffer.jl

```diff
@@ -4,21 +4,26 @@ New items are pushed to the back of the list, overwriting values in a circular f
 mutable struct CircularBuffer{T} <: AbstractVector{T}
     capacity::Int
     first::Int
+    length::Int
     buffer::Vector{T}
 
-    CircularBuffer{T}(capacity::Int) where {T} = new{T}(capacity, 1, T[])
+    CircularBuffer{T}(capacity::Int) where {T} = new{T}(capacity, 1, 0, Vector{T}(capacity))
 end
 
 function Base.empty!(cb::CircularBuffer)
-    cb.first = 1
-    empty!(cb.buffer)
+    cb.length = 0
+    cb
 end
 
 function _buffer_index(cb::CircularBuffer, i::Int)
-    n = length(cb)
-    if i < 1 || i > n
-        throw(BoundsError("CircularBuffer out of range. cb=$cb i=$i"))
+    if i < 1 || i > cb.length
+        throw(BoundsError(cb, i))
     end
+    _mod_buffer_index(cb, i)
+end
+
+function _mod_buffer_index(cb::CircularBuffer, i::Int)
```

Comment:
I don't like this function name, thre must be a better name than that.


Maybe `_buffer_index`, and then the current `_buffer_index` could be renamed `_buffer_index_checked`

Reply:
I agree, `_buffer_index_checked` and `_buffer_index` would be better names.